### PR TITLE
Thread-dependent resource loading bug fix

### PR DESCRIPTION
**Pull request by KaikiKM**

### CODE REVIEW

**Comment by DanySK** [comment]
So this basically allows for initializing different resource sets when a program is initialized by multiple threads. Looks good to me. @jakebeal, any concern?

File: protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java [hunk 1]
```diff
@@ -167,10 +185,7 @@ private ProtelisLoader() {
      * @throws IllegalArgumentException
      *             when the program has errors
      */
-    public static ProtelisProgram parse(final String program) {
-        if (Objects.requireNonNull(program, "null is not a valid Protelis program, not a valid Protelis module").isEmpty()) {
-            throw new IllegalArgumentException("The empty string is not a valid program, nor a valid module name");
```
Comment (jakebeal) [inline protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java]: Why are we losing this exception information?

Comment (DanySK) [reply protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java]: I see that null is checked for at line 188, but I concur with Jake we are losing information on empty strings. @KaikiKM, can you fix this?

Comment (KaikiKM) [reply protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java]: Sorry i probably removed it by mistake, fixed

File: protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java [hunk 2]
```diff
@@ -167,7 +185,7 @@ private ProtelisLoader() {
      * @throws IllegalArgumentException
      *             when the program has errors
      */
-    public static ProtelisProgram parse(final String program) {
+    public static ProtelisProgram parse(final String program) throws IllegalArgumentException {
```
Comment (DanySK) [inline protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java]: Remove the useless `throws`: `RuntimeException`s only need to get documented ad the Javadoc level, they should not be declared in a throws block.

Comment (KaikiKM) [reply protelis/protelis-interpreter/src/main/java/org/protelis/lang/ProtelisLoader.java]: fixed

**Comment by DanySK** [comment]
Good to go

**Pull request merged**